### PR TITLE
fix: loading tags

### PR DESCRIPTION
**Pull request by joschkabraun**
- [x] fixes issue that `json` files are not loaded when listing files from S3
- [x] speeds up loading of tags from S3 & disk

Also,
- removes unused function `_download_from_s3_bucket ` (implemented in `NOWPreprocessor`)
- fixes missing `traversal_path` to `text-to-video` app flows

### CODE REVIEW

File: now/apps/text_to_video/flow-video-clip-high-performance.yml [hunk 1]
```diff
@@ -48,7 +50,6 @@ executors:
       index_traversal_paths: '@c'
       search_traversal_paths: '@c'
       dim: ${{ ENV.PRE_TRAINED_EMBEDDINGS_SIZE }}
-      columns: ${{ ENV.COLUMNS }}
```
Comment (azayz) [inline now/apps/text_to_video/flow-video-clip-high-performance.yml:51]: why is this being deleted ? are we using only docarray indexer or we don't expect to do filtering in this case ?

File: now/data_loading/data_loading.py [hunk 2]
```diff
@@ -96,6 +108,7 @@ def merge_documents(user_input, content_file, json_file):
             tags = _open_json(json_file)
     else:
         tags = {}
+    tags = flatten_dict(tags)
```
Comment (azayz) [inline now/data_loading/data_loading.py:111]: the news tags files contain nested dictionaries ?

File: now/data_loading/data_loading.py [hunk 3]
```diff
@@ -116,17 +129,24 @@ def _load_tags_from_json(da, user_input):
         folder = d.uri.rsplit('/', 1)[0]
         folder_to_files[folder].append(d.uri)
     merged_documents = []
-    for _, files in folder_to_files.items():
-        document = get_document(files, user_input)
-        if document:
-            merged_documents.append(document)
+    with ThreadPoolExecutor() as thread_executor:
```
Comment (azayz) [inline now/data_loading/data_loading.py:132]: I assume this is for speeding things up ?

File: now/apps/text_to_video/flow-video-clip-high-performance.yml (hunk 1 above)
Comment (joschkabraun) [reply now/apps/text_to_video/flow-video-clip-high-performance.yml:51]: Good catch, wasn't supposed to be there. Thanks!

File: now/data_loading/data_loading.py (hunk 2 above)
Comment (joschkabraun) [reply now/data_loading/data_loading.py:111]: Yes, they still do. I think we should have this solution in place for the general case. If the `tags` are 'flat', then they won't be changed.

File: now/data_loading/data_loading.py (hunk 3 above)
Comment (joschkabraun) [reply now/data_loading/data_loading.py:132]: Yes

File: now/apps/text_to_video/flow-video-clip-high-performance.yml [hunk 4]
```diff
@@ -30,13 +30,15 @@ executors:
...
       traversal_paths: '@c'
+      access_paths: '@c'
       name: ${{ ENV.PRE_TRAINED_MODEL_NAME }}
     env:
       JINA_LOG_LEVEL: DEBUG
   - name: postprocessor
     uses: jinahub+docker://NOWPostprocessor/v0.0.5
     uses_with:
       traversal_paths: '@c'
+      access_paths: '@c'
```
Comment (makram93) [inline now/apps/text_to_video/flow-video-clip-high-performance.yml:41]: I don't know about this but why do we have/need two paths arguments?

**Review by makram93 — APPROVED** [review]
LGTM!

File: now/apps/text_to_video/flow-video-clip-high-performance.yml (hunk 4 above)
Comment (joschkabraun) [reply now/apps/text_to_video/flow-video-clip-high-performance.yml:41]: Yes, some executors have adopted to new naming and some don't.

**Pull request merged**